### DISCUSSION TITLE
Support a rebaseLabel that is not removed after rebase

### DISCUSSION THREAD

**Discussion by mheffner** [renovatebot/renovate#27328] [Suggest an Idea]
### Tell us more.

We use [Kodiak](https://kodiakhq.com/) to manage merge queues, but find it's challenging to get working alongside renovate. We like both tools and they perform well, but dialing in the configuration between the two is difficult.

Ideally we would like:
* renovate will rebase PRs that become conflicted
* renovate won't rebase PRs that simply fall behind main, as this can generate a lot of GHA runs ($$)
* kodiak will automerge PRs, including renovate ones, that have the `automerge` label and are ready (up-to-date, not conflicted, status checks green)
* a human must approve all PRs (no auto-merging) by adding the `automerge` label

We have configured kodiak to ignore updating PRs opened by renovate, since we found they would conflict. However, if a renovate PR is marked as `automerge` , is up to date and passes checks, then Kodiak will merge it.

We've configured renovate as such. The goal is to avoid continual rebases on renovate PRs that simply fall behind main *unless* they've been marked ready to merge with the `automerge` label.

```json
  "rebaseLabel": "automerge",
  "rebaseWhen": "conflicted",
```

However, in this config renovate will remove the `automerge` tag once it has rebased. This prevents Kodiak from picking up the PR to merge once it's ready.

It feels like having an option to rebase when a label is present and the PR is behind main, but not remove that label, would be the solution here. Also looking for guidance on other ways to achieve the goals above.

**Comment by rarkins** [comment]
Use a different label for each bot. Renovate's is configurable

**Reply by mheffner** [reply]
Hey @rarkins. I believe what we are running into is less an issue with the particular label and more the behavior. The `rebaseLabel` behavior, from what I can tell, is a one-shot notification to rebase the PR. In our scenario what we'd like to see is that once a label has been added to a PR (or several PRs) to signify it is ready to be merged, the behavior for the PR moves from `rebaseWhen:conflicted` -> `rebaseWhen:auto`. In our case this label is `automerge` since the single label works with kodiak too, but it could be a separate label as well. In either case, we would want the label to stay on the PR so that it continues to be rebased as needed until merged.

**Reply by rarkins** [reply]
How would renovate know not to keep rebasing every time it runs? This label means "manual rebase requested, just do it"

**Reply by mheffner** [reply]
That's fair, maybe the capability we're looking for isn't specifically in the existing `rebaseLabel` configuration option since that has a distinct purpose. I'm partially asking if there's a way to unlock the behavior we are looking for or, if not, whether it would be a useful capability to unlock with some future feature addition.

**Reply by rarkins** [reply]
I think you're after the equivalent of "rebase if behind base branch" label?

**Reply by mheffner** [reply]
Yes, I believe that would support our scenario.

**Reply by rarkins** [reply]
Are you willing to implement it? Because it's otherwise not a feature we'd implement ourselves. It could be called "keepUpdatedLabel"

**Reply by mheffner** [reply]
Yes I'll try to take a stab at this. Thanks for confirming this mode of operation is not something supported today and that it wasn't just us holding it wrong.

**Comment by mheffner** [comment]
This was shipped in https://github.com/renovatebot/renovate/issues/27542 and released in [37.222.0](https://github.com/renovatebot/renovate/releases/tag/37.222.0). Thanks for helping release this!
